src/Vulkan/Extensions/Silk.NET.Vulkan.Extensions.ImGui/ImGuiController.cs

```diff
@@ -0,0 +1,435 @@
+		}
+
+		private void UpdateImGuiInput()
+		{
+			var io = ImGuiNET.ImGui.GetIO();
+
+			var mouseState = _input.Mice[0].CaptureState();
+			var keyboardState = _input.Keyboards[0];
+
+			io.MouseDown[0] = mouseState.IsButtonPressed(MouseButton.Left);
+			io.MouseDown[1] = mouseState.IsButtonPressed(MouseButton.Right);
+			io.MouseDown[2] = mouseState.IsButtonPressed(MouseButton.Middle);
+
+			var point = new Point((int)mouseState.Position.X, (int)mouseState.Position.Y);
+			io.MousePos = new Vector2(point.X, point.Y);
+
+			var wheel = mouseState.GetScrollWheels()[0];
+			io.MouseWheel = wheel.Y;
+			io.MouseWheelH = wheel.X;
+
+			foreach (Key key in Enum.GetValues(typeof(Key)))
+			{
+				if (key == Key.Unknown)
+				{
+					continue;
+				}
+				io.KeysDown[(int)key] = keyboardState.IsKeyPressed(key);
+			}
+
+			foreach (var c in _pressedChars)
+			{
+				io.AddInputCharacter(c);
+			}
+
+			_pressedChars.Clear();
+
+			io.KeyCtrl = keyboardState.IsKeyPressed(Key.ControlLeft) || keyboardState.IsKeyPressed(Key.ControlRight);
+			io.KeyAlt = keyboardState.IsKeyPressed(Key.AltLeft) || keyboardState.IsKeyPressed(Key.AltRight);
+			io.KeyShift = keyboardState.IsKeyPressed(Key.ShiftLeft) || keyboardState.IsKeyPressed(Key.ShiftRight);
+			io.KeySuper = keyboardState.IsKeyPressed(Key.SuperLeft) || keyboardState.IsKeyPressed(Key.SuperRight);
+		}
+
+		internal void PressChar(char keyChar)
+		{
+			_pressedChars.Add(keyChar);
+		}
+
+		private static void SetKeyMappings()
+		{
+			var io = ImGuiNET.ImGui.GetIO();
+			io.KeyMap[(int)ImGuiKey.Tab] = (int)Key.Tab;
+			io.KeyMap[(int)ImGuiKey.LeftArrow] = (int)Key.Left;
+			io.KeyMap[(int)ImGuiKey.RightArrow] = (int)Key.Right;
+			io.KeyMap[(int)ImGuiKey.UpArrow] = (int)Key.Up;
+			io.KeyMap[(int)ImGuiKey.DownArrow] = (int)Key.Down;
+			io.KeyMap[(int)ImGuiKey.PageUp] = (int)Key.PageUp;
+			io.KeyMap[(int)ImGuiKey.PageDown] = (int)Key.PageDown;
+			io.KeyMap[(int)ImGuiKey.Home] = (int)Key.Home;
+			io.KeyMap[(int)ImGuiKey.End] = (int)Key.End;
+			io.KeyMap[(int)ImGuiKey.Delete] = (int)Key.Delete;
+			io.KeyMap[(int)ImGuiKey.Backspace] = (int)Key.Backspace;
+			io.KeyMap[(int)ImGuiKey.Enter] = (int)Key.Enter;
+			io.KeyMap[(int)ImGuiKey.Escape] = (int)Key.Escape;
+			io.KeyMap[(int)ImGuiKey.A] = (int)Key.A;
+			io.KeyMap[(int)ImGuiKey.C] = (int)Key.C;
+			io.KeyMap[(int)ImGuiKey.V] = (int)Key.V;
+			io.KeyMap[(int)ImGuiKey.X] = (int)Key.X;
+			io.KeyMap[(int)ImGuiKey.Y] = (int)Key.Y;
+			io.KeyMap[(int)ImGuiKey.Z] = (int)Key.Z;
+		}
+
+		private unsafe void RenderImDrawData(in ImDrawDataPtr drawDataPtr, in CommandBuffer commandBuffer, in Framebuffer framebuffer, in Extent2D swapChainExtent)
+		{
+			int framebufferWidth = (int)(drawDataPtr.DisplaySize.X * drawDataPtr.FramebufferScale.X);
+			int framebufferHeight = (int)(drawDataPtr.DisplaySize.Y * drawDataPtr.FramebufferScale.Y);
+			if (framebufferWidth <= 0 || framebufferHeight <= 0)
+				return;
```

Comment:
Fix branchless ifs in whole file. Generally, I prefer to have branches even for single-line ifs mainly to eliminate the potential for bugs.